editor/project_manager.cpp

```diff
@@ -109,245 +110,283 @@ class ProjectDialog : public ConfirmationDialog {
+				String n = d->get_next();
+				while (!n.is_empty()) {
+					if (!n.begins_with(".")) {
+						// Allow `.`, `..` (reserved current/parent folder names)
+						// and hidden files/folders to be present.
+						// For instance, this lets users initialize a Git repository
+						// and still be able to create a project in the directory afterwards.
+						is_folder_empty = false;
+						break;
+					}
+					n = d->get_next();
 				}
-				n = d->get_next();
-			}
-			d->list_dir_end();
+				d->list_dir_end();
 
-			if (!is_folder_empty) {
-				if (valid_path == OS::get_singleton()->get_environment("HOME") || valid_path == OS::get_singleton()->get_system_dir(OS::SYSTEM_DIR_DOCUMENTS) || valid_path == OS::get_singleton()->get_executable_path().get_base_dir()) {
-					set_message(TTR("You cannot save a project in the selected path. Please make a new folder or choose a new path."), MESSAGE_ERROR);
-					get_ok_button()->set_disabled(true);
+				if (!is_folder_empty) {
+					set_message(TTR("The selected path is not empty. Choosing an empty folder is highly recommended."), MESSAGE_WARNING, target_path_type);
 					return "";
 				}
-
-				set_message(TTR("The selected path is not empty. Choosing an empty folder is highly recommended."), MESSAGE_WARNING);
-				get_ok_button()->set_disabled(false);
-				return valid_path;
+			} else {
+				is_folder_empty = true;
 			}
 		}
 
-		set_message("");
-		set_message("", MESSAGE_SUCCESS, INSTALL_PATH);
-		get_ok_button()->set_disabled(false);
-		return valid_path;
+		return path;
 	}
 
-	void _path_text_changed(const String &p_path) {
-		String sp = _test_path();
-		if (!sp.is_empty()) {
-			// If the project name is empty or default, infer the project name from the selected folder name
-			if (project_name->get_text().strip_edges().is_empty() || project_name->get_text().strip_edges() == TTR("New Game Project")) {
-				sp = sp.replace("\\", "/");
-				int lidx = sp.rfind("/");
+	// Sanitizes the project name or import path to be used as a directory name.
+	String _get_suggested_directory() {
+		String name;
+		if (mode == MODE_NEW) {
+			name = project_name->get_text();
+		} else if (mode == MODE_IMPORT) {
+			name = project_path->get_text().get_file().get_basename();
+		} else {
+			ERR_FAIL_V_MSG("", "Invalid project mode " + String(Variant(mode)) + ".");
+		}
+		int naming_convention = (int)EDITOR_GET("project_manager/directory_naming_convention");
+		switch (naming_convention) {
+			case 0: // No convention
+				break;
+			case 1: // kebab-case
+				name = name.to_snake_case().replace("_", "-");
+				break;
+			case 2: // snake_case
+				name = name.to_snake_case();
+				break;
+			case 3: // camelCase
+				name = name.to_camel_case();
+				break;
+			case 4: // PascalCase
+				name = name.to_pascal_case();
+				break;
+			case 5: // Title Case
+				name = name.capitalize();
+				break;
+			default:
+				ERR_FAIL_V_MSG("", "Invalid directory naming convention.");
+				break;
+		}
```

Comment:
Open to better ideas

Also, right now project name `test3` has path `test-3` which I don't really like

Reply:
Still not totally satisfied with this. `kebab-case` as a default makes sense though, e.g. `godot-proposals`.